libs/vkd3d/command.c

```diff
@@ -8802,6 +8810,68 @@ static struct d3d12_command_list *unsafe_impl_from_ID3D12CommandList(ID3D12Comma
     return CONTAINING_RECORD(iface, struct d3d12_command_list, ID3D12GraphicsCommandList_iface);
 }
 
+static inline struct d3d12_command_list *d3d12_command_list_from_ID3D12GraphicsCommandListExt(ID3D12GraphicsCommandListExt *iface)
+{
+    struct d3d12_command_list_ext* list_ext = CONTAINING_RECORD(iface, struct d3d12_command_list_ext, ID3D12GraphicsCommandListExt_iface);
+    return CONTAINING_RECORD(list_ext, struct d3d12_command_list, command_list_ext);
+}
+
+static ULONG STDMETHODCALLTYPE d3d12_command_list_ext_AddRef(ID3D12GraphicsCommandListExt *iface)
+{
+    struct d3d12_command_list *command_list = d3d12_command_list_from_ID3D12GraphicsCommandListExt(iface);
+    return ID3D12GraphicsCommandList_AddRef(&command_list->ID3D12GraphicsCommandList_iface);
+}
+
+static ULONG STDMETHODCALLTYPE d3d12_command_list_ext_Release(ID3D12GraphicsCommandListExt *iface)
+{
+    struct d3d12_command_list *command_list = d3d12_command_list_from_ID3D12GraphicsCommandListExt(iface);
+    return ID3D12GraphicsCommandList_Release(&command_list->ID3D12GraphicsCommandList_iface);
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_command_list_ext_QueryInterface(ID3D12GraphicsCommandListExt *iface,
+        REFIID iid, void **out)
+{
+    TRACE("iface %p, iid %s, out %p.\n", iface, debugstr_guid(iid), out);
+
+    if (IsEqualGUID(iid, &IID_ID3D12GraphicsCommandListExt) ||
+        IsEqualGUID(iid, &IID_IUnknown))
+    {
+        d3d12_command_list_ext_AddRef(iface);
+        *out = iface;
+        return S_OK;
+    }
+
+    WARN("%s not implemented, returning E_NOINTERFACE.\n", debugstr_guid(iid));
+
+    *out = NULL;
+    return E_NOINTERFACE;
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_command_list_ext_GetDevice(ID3D12GraphicsCommandListExt *iface,
+        REFIID iid, void **out)
+{
+    struct d3d12_command_list *command_list = d3d12_command_list_from_ID3D12GraphicsCommandListExt(iface);
+    return ID3D12CommandList_GetDevice(&command_list->ID3D12GraphicsCommandList_iface, iid, out);
```

Comment:
d3d12_command_list_GetDevice

Reply:
Also not sure why this is useful. You could just have called GetDevice directly from ID3D12GraphicsCommandList instead of chaining via the Ext interface